src/main/java/com/zegoggles/smssync/preferences/Preferences.java

```diff
@@ -161,8 +163,17 @@ public int getRegularTimeoutSecs() {
         return getStringAsInt(REGULAR_TIMEOUT_SECONDS, Defaults.REGULAR_TIMEOUT_SECONDS);
     }
 
-    public boolean getMarkAsRead() {
-        return preferences.getBoolean(MARK_AS_READ.key, Defaults.MARK_AS_READ);
+    public void migrateMarkAsRead() {
+        if (preferences.contains(MARK_AS_READ.key)) {
+            if (!preferences.getBoolean(MARK_AS_READ.key, true)) {
+                preferences.edit().putString(MARK_AS_READ_TYPES.key, "unread").commit();
```

Comment:
i know it would work without it, but since you check `preferences.contains(MARK_AS_READ.key)` previously you know that the user has already set something explictly, so should also be converted to a value for consistency